cmd/control_loop/controller/controller.go

```diff
@@ -24,17 +25,29 @@ func ProcessManifest(manifest *Manifest) ([]string, error) {
 		if err != nil {
 			return nil, err
 		}		
-		resourcePattern := fmt.Sprintf("projects/%s/%s", manifest.Project, entry.Resource)
+		skipEntry := false
 
+		// Generate dependency map
+		resourcePattern := fmt.Sprintf("projects/%s/%s", manifest.Project, entry.Resource)
 		dependencyMaps := make([]map[string]*ResourceCollection, 0, 0)
-		err = populateDependencyMaps(ctx, client, resourcePattern, entry.Dependencies, &dependencyMaps)
-		if err != nil {
-			return nil, err
+		for _, d := range entry.Dependencies {
+			dMap, err := generateDependencyMap(ctx, client, resourcePattern, d.Source, d.Filter)
+			if err != nil {
+				log.Printf("Encountered error during generateDependencyMap().\n Error: %s\n Skipping entry %+v", err.Error(), entry)
```

Comment:
Thanks for this suggestion, refactored. 

I put the client declaration outside of the loop. Each iteration of the loop will now share the same registry client. The logic behind returning vs logging the errors is that, if there is an error in processing a particular manifest entry, we should log those  since we don't want to return and skip processing the remaining entries which might be valid. 

Errors like client creation should return because, it doesn't make sense to keep retrying client creation for every iteration, if we know it failed the first time. Now that the client creation is outside of the loop, the for log will only log errors.